### PR TITLE
CCIP-2160 : diff msgtype in load profile

### PR DESCRIPTION
**Pull request by AnieeG**
## Solution

- Group all load related configs in to one struct
- Enable sending multiple msg type in load run with their corresponding frequency. For example you can specify the following Msg types by  providing the frequency as [4, 12, 3, 1]

````
20%: Token Transfers - 

data.length = 0Kb
gasLimit = 0

60%: Programmable Token Transfers -

data.length = 5Kb
gasLimit = 500K

15%: Messaging - 

data.length = 10Kb
gasLimit = 800K

5%: Messaging - 

data.length = 10Kb
gasLimit = 2.5M
`
```

### CODE REVIEW

File: integration-tests/ccip-tests/testconfig/tomls/scalability.toml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+[CCIP.Groups.load.LoadProfile]
+TestDuration = '4h'
+TimeUnit = '5s'
+RequestPerUnitTime = [1]
+OptimizeSpace = true
+SendMaxDataInEveryMsgCount = 1000
+
+# to represent 20%, 60%, 15%, 5% of the total messages
+[CCIP.Groups.load.LoadProfile.MsgProfile]
+Frequencies = [4,12,3,1]
```
Comment (mateusz-sekara) [inline integration-tests/ccip-tests/testconfig/tomls/scalability.toml:96]: Some docs on how to adjust these numbers would be very handy. Maybe in README?

File: integration-tests/ccip-tests/testconfig/tomls/ccip-default.toml [hunk 2]
```diff
@@ -274,30 +274,52 @@ CommitAndExecuteOnSameDON = true # same as above
...
+[CCIP.Groups.load.LoadProfile]
+RequestPerUnitTime = [1] # number of ccip requests to be sent per unit time
+TimeUnit = '10s' # unit of time for RequestPerUnitTime
+TestDuration = '10m'                # load test duration, not used for smoke tests
+WaitBetweenChaosDuringLoad = '2m'   # Duration to wait between each chaos injection during load test; only valid for chaos tests
 
-[CCIP.Groups.load.MsgDetails]
+# uncomment the following if you want your test results to be reflected under CCIP test grafana dashboard with namespace label same as the value of the following variable
+# TestRunName = <env>_<testnet/mainnet>_<cciprelease> i.e prod-testnet-2.7.1-ccip1.2.1-beta
+# Message Frequency Distribution Example
```
Comment (AnieeG) [inline integration-tests/ccip-tests/testconfig/tomls/ccip-default.toml:294]: @mateusz-sekara the comment describes how to use msg frequency

File: integration-tests/ccip-tests/load/ccip_test.go [hunk 3]
```diff
@@ -108,8 +108,8 @@ func TestLoadCCIPStableRequestTriggeringWithNetworkChaos(t *testing.T) {
 			testEnv.K8Env.Cfg.Namespace, &chaos.Props{
 				FromLabels:  &map[string]*string{"geth": ptr.Ptr(actions.ChaosGroupCCIPGeth)},
 				ToLabels:    &map[string]*string{"app": ptr.Ptr("chainlink-0")},
-				DurationStr: testArgs.TestCfg.TestGroupInput.TestDuration.String(),
-				Delay:       "300ms",
+				DurationStr: testArgs.TestCfg.TestGroupInput.LoadProfile.TestDuration.String(),
+				Delay:       "200ms",
```
Comment (mateusz-sekara) [inline integration-tests/ccip-tests/load/ccip_test.go:112]: For the reference, probably something that should be configurable as well ;)

File: integration-tests/ccip-tests/testconfig/tomls/ccip-crib.toml [hunk 4]
```diff
@@ -1,163 +1,94 @@
...
 evm_default_gas_limit = 6000000
-evm_finality_depth = 1
+evm_finality_tag = true
 
 [CCIP.Env.ExistingCLCluster]
 Name = 'crib-ani'
 NoOfNodes = 6
 
 [[CCIP.Env.ExistingCLCluster.NodeConfigs]]
-URL = 'https://skudasov-crib-ccip-node1.main.stage.cldev.sh/'
```
Comment (mateusz-sekara) [inline integration-tests/ccip-tests/testconfig/tomls/ccip-crib.toml:113]: Should we keep changing these or just keep some generic values?

**Review by mateusz-sekara — APPROVED** [review]
lgtm!

File: integration-tests/ccip-tests/load/ccip_test.go (hunk 3 above)
Comment (AnieeG) [reply integration-tests/ccip-tests/load/ccip_test.go:112]: Coming up in next PR

File: integration-tests/ccip-tests/testconfig/tomls/ccip-crib.toml (hunk 4 above)
Comment (AnieeG) [reply integration-tests/ccip-tests/testconfig/tomls/ccip-crib.toml:113]: yeah should be generic values :)

**Review by skudasov — APPROVED** [review]

File: .github/workflows/integration-tests-publish.yml [hunk 5]
```diff
@@ -37,7 +37,6 @@ jobs:
           ref: ${{ github.event.pull_request.head.sha || github.sha }}
       - name: Setup Other Tags If Not Workflow Dispatch
         id: tags
-        if: github.event_name != 'workflow_dispatch'
```
Comment (kalverra) [inline .github/workflows/integration-tests-publish.yml:40]: Why remove this? Seems like the step is only for non-dispatch jobs.

File: integration-tests/ccip-tests/actions/ccip_helpers.go [hunk 6]
```diff
@@ -2876,6 +2881,7 @@ func (lane *CCIPLane) StartEventWatchers() error {
 					if sub != nil {
```
Comment (kalverra) [inline integration-tests/ccip-tests/actions/ccip_helpers.go:2881]: If we're using the `event.Resubscribe` call above to handle this logic for us, isn't this duplicating it?

**Pull request merged**